common/src/main/java/org/apache/cassandra/sidecar/common/CassandraAdapterDelegate.java

```diff
@@ -133,45 +133,46 @@ private void healthCheckInternal()
         if (activeSession == null)
         {
             logger.info("No local CQL session is available. Cassandra is down presumably.");
-            isUp = false;
+            status = null;
             return;
         }
 
         maybeRegisterHostListener(activeSession);
 
         try
         {
-            String version = activeSession.execute("select release_version from system.local")
-                                          .one()
-                                          .getString("release_version");
-            isUp = true;
+            Row oneResult = activeSession.execute("select release_version, partitioner from system.local")
+                                   .one();
+            NodeSettings nodeSettings = new NodeSettings(oneResult.getString("release_version"),
+                                                   oneResult.getString("partitioner"));
+            this.status = nodeSettings;
             // this might swap the adapter out
-            SimpleCassandraVersion newVersion = SimpleCassandraVersion.create(version);
+            SimpleCassandraVersion newVersion = SimpleCassandraVersion.create(nodeSettings.releaseVersion());
             if (!newVersion.equals(currentVersion))
             {
                 currentVersion = newVersion;
-                adapter = versionProvider.getCassandra(version).create(cqlSession);
+                adapter = versionProvider.getCassandra(nodeSettings.releaseVersion()).create(cqlSession);
                 logger.info("Cassandra version change detected. New adapter loaded: {}", adapter);
             }
-            logger.debug("Cassandra version {}", version);
+            logger.debug("Cassandra version {}", nodeSettings.releaseVersion());
         }
         catch (NoHostAvailableException e)
         {
             logger.error("Unexpected error connecting to Cassandra instance.", e);
             // The cassandra node is down.
             // Unregister the host listener and nullify the session in order to get a new object.
-            isUp = false;
+            status = null;
             maybeUnregisterHostListener(activeSession);
             session = null;
```

Comment:
On a different note, doesn't it make sense to close the session in this case?

Reply:
@nvharikrishna I think we want to keep the session open as long as the node is running, @frankgh  correct me if I am wrong.

Reply:
At this line existing session is assigned with null (throwing away the existing session). Thought it is better to close before assigning it to null.